### ISSUE TITLE
Some messages reaching outside of human space

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#2296]
So I was redoing the FW campaign and while hanging out in Coalition space during one of the breaks in the story, I received a message from Freya about Stack Cores being ready. Is it intentional that messages like that are able to reach the player when outside of human space or is this a mistake, because I believe some other messages can only be received while in human space.

I'm not entirely sure how many times this comes up, but this is the first one I've witnessed.

**Comment by Bladewood** [comment]
This is probably a mistake. Humans rely on the hyperspace relays to transmit their messages. Given that there are no such relays connection Coalition space to human space, we can consider this to be an error.

**Comment by gunqqer** [comment]
It is true that human communications should not be able to reach outside of human space lore wise. Except if we look at this from a gameplay perspective I don't have to have to keep going to a human planet to make sure I didn't miss something. If those messages did not appear you could miss an important mission and mess up an entire story line.

**Comment by Bladewood** [comment]
There's also the problem of several messages arriving at the same time if you delay them by going into other regions of space.
For example, the message that tells me that it is Remembrance Day and the message that tells me that the Free Worlds have declared independence can arrive one after the other in the same day if I just hide out in Hai space for a few years.

**Comment by endless-sky** [comment]
Yeah, it's a balance between realism vs. not delaying information. I kind of think breaking realism is preferable, but it would be easy to make messages like that only appear in systems that are `near Sol 1000`, i.e. hyperspace-connected to human space, if people think that's preferable.

**Comment by Jorlem** [comment]
Would it be possible to compromise by having different text for the events, if you are not in human space? As I recall, there are humans living in Hai space, so major events in human space might be mentioned on Hai news services for said humans, or in other  areas it could focus on the actions of other aliens, like the Pug acting against the humans, instead of being from a human perspective.

Or, a bit could be added to the jump drive's description, so it is stated to include a transmission relay that the PC sets up on a human would, in order to stay up to date on the news, to provide an in-universe explanation for how the player can get those transmissions no matter where in the galaxy they are.

**Comment by Tadrix** [comment]
Hai space is connected to human space via wormhole from the start, so messages from human space should reach the player while there. Other places, well… probably not so much.
> near Sol 1000

Do wormholes count as a connection?

**Comment by jafdy** [comment]
no

**Comment by Wrzlprnft** [comment]
as far as i recall the messages reaching hai space are shipped over by merchant ships and they send it on the other side.

**Comment by KiLEdEnNis** [comment]
is it maybe an idea that a Quarg ship chases you & once it's in the same system ya receive the message?
kinda weird but seemed like a funny feature
(or even have a Pointedstick Vanguard w/ a jump drive deliver the message)

**Comment by Tadrix** [comment]
Yeah, because the _volunteered_ to deliver messages. :D

**Comment by Amazinite** [comment]
Children of system can't have children like that.
You would just do
```
system
	near Sol 1000
	near "Heia Due" 1000
```

**Comment by Tadrix** [comment]
Probably dumb question, but isn't what you wrote requiring the player being in **both** the vicinity of Sol and Heia Due systems?

**Comment by KiLEdEnNis** [comment]
no, if 1 of the 2 requirements is met it works
compare it to like transport missions
_source
__attributes "near earth" "dirt belt" "north"
it takes any planet w/ 1 of the 3 tags, not with all 3 tags

I think what you confused it with is misasions' "to offer"

**Comment by Amazinite** [comment]
First off, system can't have a child named near. It should have been source.
And secondly, it should probably be like this:
```
source
	near
		Sol 1000
		"Heia Due" 1000
```
But it doesn't look like near can have children like this.

**Comment by Amazinite** [comment]
This happens with the Kestrel announcement as well.

**Comment by gunqqer** [comment]
Some messages that aren't really important to the plot, like the Kestrel Announcement or the couple that you took from smugglers den should be restricted to human space with 
`system
     near
         "New Boston"`
But anything that continues are story line or is time sensitive should be given anywhere no matter what

**Comment by Amazinite** [comment]
Well none of the announcements are time sensitive. They're usually something like "come see us on wherever when you're ready."

**Comment by EndrosG** [comment]
note that it makes a difference whether you note stuff into one line or in separate lines

Why do you compare
- near Sol
- near Somewhere

with
- attributes "near earth" "dirt belt"

and not with
- attributes "near earth"
- attributes "dirt belt"

? This is in terms of game mechanics a big difference since the latter one is matched with logical AND while the one-liner is a logical OR.

**Comment by endless-sky** [comment]
Addressed in #3106.
